test/stringify.js

```diff
@@ -143,6 +143,32 @@ test('stringify()', function (t) {
         st.end();
     });
 
+    t.test('omits object key/value pair when value is empty array', function (st) {
+        st.equal(qs.stringify({ a: [], b: 'zz' }), 'b=zz');
+    });
+
+    t.test('should not omit object key/value pair when value is empty array and when asked', function (st) {
+        st.equal(qs.stringify({ a: [], b: 'zz' }), 'a[]b=zz', { allowEmptyArrays: true });
+        st.equal(qs.stringify({ a: [], b: 'zz' }), 'b=zz', { allowEmptyArrays: false });
+    });
+
+    t.test('should throw when allowEmptyArrays is not of type boolean', function (st) {
+        st['throws'](function () {
+            qs.stringify({ a: [], b: 'zz' }, { allowEmptyArrays: 'foobar' });
+        }, new TypeError('allowEmptyArrays option property can only be true or false'));
+        st['throws'](function () {
+            qs.stringify({ a: [], b: 'zz' }, { allowEmptyArrays: 0 });
+        }, new TypeError('allowEmptyArrays option property can only be true or false'));
+        st['throws'](function () {
+            qs.stringify({ a: [], b: 'zz' }, { allowEmptyArrays: NaN });
+        }, new TypeError('allowEmptyArrays option property can only be true or false'));
```

Comment:
```suggestion

        st['throws'](
            function () { qs.stringify({ a: [], b: 'zz' }, { allowEmptyArrays: NaN }); },
            TypeError
        );
```
same changes throughout, including the newlines between